pages/time-bomb.tsx

```diff
@@ -0,0 +1,207 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { CenteredMain } from '../components/Centered';
+import { useRouter } from 'next/router';
+import { checkAnswer } from '../services/checkAnswer';
+import Countdown, { zeroPad } from 'react-countdown';
+
+type MainColor = 'red' | 'green' | 'blue';
+type ExtendedColor = MainColor | 'gray';
+
+interface ICountdownRenderer {
+    minutes: number;
+    seconds: number;
+    milliseconds: number;
+    completed: boolean;
+}
+
+interface IColorButton {
+    originalColor: MainColor;
+    onClick: () => void;
+}
+
+interface IBomb {
+    onCountdownComplete: () => void;
+    start: boolean;
+    endTime: number | null;
+}
+
+const LOCK_LETTER = 'T';
+
+const RED = 'red';
+const GREEN = 'green';
+const BLUE = 'blue';
+
+const ControlPanelGrid = styled.div`
+    background: gray;
+    display: grid;
+    grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr;
+    grid-template-rows: 1fr 1fr 1fr 1fr 1fr 1fr;
+    column-gap: 10px;
+    row-gap: 10px;
+    height: 600px;
+    width: 600px;
+    border: 5px solid black;
+`;
+
+const ColorSquare = styled.div`
+    background: ${({ color }) => color};
+    border: 1px solid black;
+`;
+
+const ColorButton: React.FC<IColorButton> = ({ originalColor, onClick }) => {
+  const [color, setColor] = useState<ExtendedColor>(originalColor);
+
+  const handleClick = () => {
+    onClick();
+    setColor('gray');
+  };
+
+  return <ColorSquare onClick={handleClick} color={color} />;
+};
+
+function getRandomColor(): MainColor {
+  const colors: MainColor[] = [RED, GREEN, BLUE];
+  return colors[Math.floor(Math.random() * colors.length)];
```

Comment:
I want to make a helper function soon to get a random member of an array, since it is something we do alot.